Split the Extensions page in WP Admin and add respective menu items

madeincosmos opened the pull request:
Continuation of #30378 so that phpcs errors on the My Subscriptions page aren't a blocker for now.

### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:
- Register submenu for My Subscriptions under WooCommerce
- Add a method in `WC_Admin_Addons` for Helper page output only
- remove top nav from the Extensions page (keeping it on the My Subscriptions page for now so phpcs is not a blocker)
- update e2e tests for Helper connection.
<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

### How to test the changes in this Pull Request:

1. Check out this branch on local environment.
2. Go to WP Admin and observe there are two separate submenus for Extensions and My Subscriptions.
3. Open each of these pages and make sure they load correctly and have all styles in place.
4. Make sure the correct menu item is highlighted when you open each page.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Screenshots
![](https://d.pr/i/55j3Qb+)

### Notes
- didn't remove top nav on the My Subscriptions page yet because editing this file triggered a lot of phpcs errors. I'll take care of it later.

Kleindberg committed:
Added Ukrainian states

https://en.wikipedia.org/wiki/Oblasts_of_Ukraine

YordanSoares committed:
Fix the payment method title for the option "Other"

Closes: #30255

danbitzer committed:
Set $object_type prop for WC_Customer class

danbitzer committed:
Remove \WC_Customer::get_hook_prefix method

This method is now redundant since the `$object_type` property is set for the class. This means the parent method will now return the exact same value.

zhongruige committed:
Removed ignore lint fields

zhongruige committed:
Update README

madeincosmos committed:
Split Extensions page into 2 menu items.

madeincosmos committed:
Update e2e tests.

roykho committed:
Allow user with capabilities to view downloadable image closes #29988

kloon commented on includes/admin/class-wc-admin-menus.php:
```diff
@@ -151,6 +152,7 @@ public function addons_menu() {
 		/* translators: %s: extensions count */
 		$menu_title = sprintf( __( 'Extensions %s', 'woocommerce' ), $count_html );
```
Let's use this variable for My Subscriptions menu item, as it contains the subscription update count.

madeincosmos replied:
Thanks, updated it now 🙌

kloon commented on includes/admin/class-wc-admin-menus.php:
```diff
@@ -151,6 +152,7 @@ public function addons_menu() {
 		/* translators: %s: extensions count */
 		$menu_title = sprintf( __( 'Extensions %s', 'woocommerce' ), $count_html );
 		add_submenu_page( 'woocommerce', __( 'WooCommerce extensions', 'woocommerce' ), $menu_title, 'manage_woocommerce', 'wc-addons', array( $this, 'addons_page' ) );
```
Title of this menu needs to be `Marketplace`

madeincosmos replied:
Renamed to Marketplace.

kloon reviewed (approved):
Changes look good, just left two things to change before we merge into the feature branch.

andfinally reviewed (commented):
Looking great, thanks! I think we need to make some additional changes to the My Subscriptions page, http://localhost:8100/wp-admin/admin.php?page=wc-addons&section=helper. Apologies for not spelling these out in the issue.

* The page title at the top of the page should be "My Subscriptions" instead of "Extensions".
* We should remove the tab bar on this page, as the "Browse Extensions" page is now the separate Marketplace page.
* We should replace the h2 "Subscriptions" with "WooCommerce.com Subscriptions".

Also, on the Marketplace page, the page title at the top needs to be Marketplace, as well as the menu item Gerhard mentioned.

<img width="500" alt="image" src="https://user-images.githubusercontent.com/1647564/127627362-5080ca63-1ddc-417c-a479-078301c44ec1.png">

<img width="500" alt="image" src="https://user-images.githubusercontent.com/1647564/127627534-02e42c07-bd0b-441d-a0ed-7f51e81d3afa.png">

madeincosmos committed:
Change menu item name to Marketplace.

madeincosmos committed:
Change My Subscriptions page header.

kloon commented:
The titles need to be adjusted in WC Admin

https://github.com/woocommerce/woocommerce-admin/blob/38fbb4aedd0f21cf21ddaa6af93e48564fd35324/includes/connect-existing-pages.php#L24

Konamiman committed:
Extend post-merge GH action to add a comment suggestion next steps.

Konamiman committed:
Merge pull request #30385 from woocommerce/fix/29988

Allow user with capabilities to view downloadable image closes #29988

Konamiman committed:
Refactor the PR post-merge processing GitHub action

Common code has been extracted to a "post-request-shared.php" script.

madeincosmos commented:
All feedback on this PR is addressed now except for Marketplace and My Subscriptions page headers than need to be set in WooCommerce Admin instead. 

I'm going to merge this one once the PHPCS issues are addressed in #30405.

barryhughes committed:
Merge pull request #30408 from woocommerce/add/29852

GitHub action to post a comment with next steps suggestion on pull request merge.

claudiosanches committed:
Update 5.6 changelog

barryhughes committed:
Merge pull request #30413 from woocommerce/update/changelog

Update 5.6 changelog.

Konamiman committed:
Fix: the same set of related products is always displayed

get_related_products_query in the WC_Product_Data_Store_CPT class is
querying the suitable products with a "limit" clause, this causes
that products down in the table are never picked for display.
This commit adds a ORDER BY RAND() to the query so that all products
have a chance to get picked.

roykho committed:
Merge pull request #30418 from woocommerce/fix/30140

Fix: "Related Products" not showing all random related products within the same category

madeincosmos committed:
Cleanup unused functions and files.

madeincosmos committed:
Fix phpcs issues.

andfinally committed:
Merge branch 'update/marketplace-1' into update/split-extensions-pages

vedanshujain committed:
Merge pull request #30325 from woocommerce/add/wc-customer-data-type-prop-30272

Specify the `$object_type` property for the `WC_Customer` class

kloon commented on includes/admin/views/html-admin-page-addons.php:
```diff
@@ -13,6 +13,8 @@
 
 ?>
 <div class="wrap woocommerce wc_addons_wrap">
+	<h1 class="screen-reader-text"><?php esc_html_e( 'Marketplace', 'woocommerce' ); ?></h1>
```
Let's change this to `WooCommerce Marketplace` to match the title, this is only for screen readers but should match the main page title.

madeincosmos replied:
Thanks, I updated the title now.

kloon commented on includes/admin/class-wc-admin-menus.php:
```diff
@@ -149,8 +150,9 @@ public function status_menu() {
-		$menu_title = sprintf( __( 'Extensions %s', 'woocommerce' ), $count_html );
-		add_submenu_page( 'woocommerce', __( 'WooCommerce extensions', 'woocommerce' ), $menu_title, 'manage_woocommerce', 'wc-addons', array( $this, 'addons_page' ) );
+		$menu_title = sprintf( __( 'My Subscriptions %s', 'woocommerce' ), $count_html );
+		add_submenu_page( 'woocommerce', __( 'WooCommerce extensions', 'woocommerce' ), __( 'Marketplace', 'woocommerce' ), 'manage_woocommerce', 'wc-addons', array( $this, 'addons_page' ) );
```
```suggestion
		add_submenu_page( 'woocommerce', __( 'WooCommerce Marketplace', 'woocommerce' ), __( 'Marketplace', 'woocommerce' ), 'manage_woocommerce', 'wc-addons', array( $this, 'addons_page' ) );
```
Let's change the browser title to match the actual title on the page.

madeincosmos replied:
Updated ✅

kloon reviewed (approved):
Latest changes works great, thanks @madeincosmos.

I am approving assume you can just address the two small title changes so we have consistent titles.

zhongruige committed:
Updated changelog

zhongruige committed:
Merge pull request #30361 from woocommerce/add/e2e-shipping-classes-cleanup

Added shipping classes cleanup to retest flow

Kleindberg committed:
Update states.php

Kleindberg committed:
Update states.php

andfinally reviewed (approved):
Thanks for this excellent, neat solution! 👌

Konamiman committed:
Merge pull request #30256 from YordanSoares/issue-30255-fix-payment-method-title-for-the-option-other

Fix the payment method title for the option "Other"

andfinally commented:
For some reason I'm still getting "WooCommerce extensions" as the browser tab title on both pages, I think this may be related to the issue you [referred](p1628245850167600-slack-C01LE21AWCD) to.

Konamiman committed:
Merge pull request #30230 from Kleindberg/patch-1

Added Ukrainian states

madeincosmos committed:
Merge branch 'update/split-extensions-pages' of github.com:woocommerce/woocommerce into update/split-extensions-pages

madeincosmos committed:
Update Marketplace and My Subscriptions page titles.

andfinally committed:
Correcting escaping in the subscriptions list in `html-main.php` – `esc_html` was escaping a `strong` tag in the renewal message.
